zipread/reader.go

```diff
@@ -175,6 +178,44 @@ func (f *File) Open() (io.ReadCloser, error) {
 	}, nil
 }
 
+// OpenAsGzip returns a ReadCloser that provides access to the File's compressed contents.
+// This method returns an ErrAlgorithm error if the zip is not compressed using deflate.
+func (f *File) OpenAsGzip() (io.ReadCloser, error) {
+	size := int64(f.CompressedSize64)
+
+	if f.Method != Deflate {
```

Comment:
this is crazy but if f.Method == Store (no compression) we could just compress, that way this method works for both defined f.Method types

Reply:
That's a good option.  I figured if this method failed, I'd just call Open() and compress the result if that's what's requested.  I haven't thought much about minimizing round trips yet, so I'll consider your approach.